backend/controllers/cellMove/selectReception.ts

```diff
@@ -0,0 +1,233 @@
+import moment from 'moment'
+import { alertFlagLabels, cellMoveAlertCodes } from '../../shared/alertFlagValues'
+
+import { putLastNameFirst, hasLength, groupBy, properCaseName, formatName, formatLocation } from '../../utils'
+
+import {
+  userHasAccess,
+  getNonAssociationsInEstablishment,
+  renderLocationOptions,
+  cellAttributes,
+  translateCsra,
+} from './cellMoveUtils'
+
+const defaultSubLocationsValue = { text: 'Select area in residential unit', value: '' }
+const noAreasSelectedDropDownValue = { text: 'No areas to select', value: '' }
+const toDropDownValue = (entry) => ({ text: entry.name, value: entry.key })
+
+const sortByDescription = (a, b) => a.description.localeCompare(b.description)
+
+const sortByLatestAssessmentDateDesc = (left, right) => {
+  const leftDate = moment(left.assessmentDate, 'DD/MM/YYYY')
+  const rightDate = moment(right.assessmentDate, 'DD/MM/YYYY')
+
+  if (leftDate.isBefore(rightDate)) return 1
+  if (leftDate.isAfter(rightDate)) return -1
+
+  return 0
+}
+
+const getCellOccupants = async (res, { prisonApi, activeCaseLoadId, cells, nonAssociations }) => {
+  const currentCellOccupants = (
+    await Promise.all(
+      cells.map((cell) => cell.id).map((cellId) => prisonApi.getInmatesAtLocation(res.locals, cellId, {}))
+    )
+  ).flatMap((occupant) => occupant)
+
+  if (!hasLength(currentCellOccupants)) return []
+
+  const occupantOffenderNos = Array.from(new Set(currentCellOccupants.map((occupant) => occupant.offenderNo)))
+
+  const occupantAlerts = await prisonApi.getAlerts(res.locals, {
+    agencyId: activeCaseLoadId,
+    offenderNumbers: occupantOffenderNos,
+  })
+  const occupantAssessments = await prisonApi.getCsraAssessments(res.locals, occupantOffenderNos)
+  const assessmentsGroupedByOffenderNo = occupantAssessments ? groupBy(occupantAssessments, 'offenderNo') : []
+
+  const cellSharingAssessments = Object.keys(assessmentsGroupedByOffenderNo)
+    .map(
+      (offenderNumber) =>
+        assessmentsGroupedByOffenderNo[offenderNumber]
+          .filter((assessment) => assessment?.assessmentDescription.includes('CSR'))
```

Comment:
this is not quite the same as previously, i.e. do you know that assessmentDescription cant be null?
Also I was suggesting you could make a similar change e.g. at lines 75, 92, 104